scripts/services/migrations.ts

```diff
@@ -1,7 +1,36 @@
-import { compat, types as T } from "../deps.ts";
+import { compat, matches, types as T } from "../deps.ts";
 
 export const migration: T.ExpectedExports.migration = compat.migrations
-  .fromMapping( {
+  .fromMapping(
+    {
+      // 25.0.2 - initial release
+      //
+      "25.0.3.3": {
+        up: compat.migrations.updateConfig(
+          async (config, effects) => {
+            if (
+              matches.shape({
+                "username": matches.unknown,
+                "password": matches.string.optional(),
+                "enable-tor": matches.unknown,
+              }).test(config)
+            ) {
+              delete config.username;
+              await effects.writeFile({
+                path: "start9/password.dat",
+                toWrite: config.password || "",
+                volumeId: "main",
+              });
+              delete config.password;
+              delete config["enable-tor"];
+            }
+            return config;
+          },
+          false, // setting to needs config due to potential bug on service update
+          { version: "25.0.3.3", type: "up" },
+        ),
+        down: () => { throw new Error('Downgrade prohibited') },
+      },
     },
-    "25.0.2" 
-);
+    "25.0.3.3",
+  );
```

Comment:
What's this for?

Reply:
my understanding is the migration.ts current version should keep in sync with the manifest version